parser/errorformat.go

```diff
@@ -0,0 +1,55 @@
+package parser
+
+import (
+	"io"
+	"strings"
+
+	"github.com/reviewdog/errorformat"
+	"github.com/reviewdog/reviewdog/proto/rdf"
+)
+
+var _ Parser = &ErrorformatParser{}
+
+// ErrorformatParser is errorformat parser.
+type ErrorformatParser struct {
+	efm *errorformat.Errorformat
+}
+
+// NewErrorformatParser returns a new ErrorformatParser.
+func NewErrorformatParser(efm *errorformat.Errorformat) *ErrorformatParser {
+	return &ErrorformatParser{efm: efm}
+}
+
+// NewErrorformatParserString returns a new ErrorformatParser from errorformat
+// in string representation.
+func NewErrorformatParserString(efms []string) (*ErrorformatParser, error) {
+	efm, err := errorformat.NewErrorformat(efms)
+	if err != nil {
+		return nil, err
+	}
+	return NewErrorformatParser(efm), nil
+}
+
+func (p *ErrorformatParser) Parse(r io.Reader) ([]*rdf.Diagnostic, error) {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
exported method ErrorformatParser.Parse should have comment or be unexported